packages/site/src/pages/index.tsx

```diff
@@ -0,0 +1,147 @@
+import {
+  Button,
+  Flex,
+  Grid,
+  Icon,
+  Input,
+  InputGroup,
+  InputLeftElement,
+  Stack,
+  Text,
+  useToast,
+} from "@chakra-ui/core";
+import copy from "copy-to-clipboard";
+import download from "downloadjs";
+import JSZip from "jszip";
+import { useEffect, useRef, useState } from "react";
+import { StringParam, useQueryParam } from "use-query-params";
+import Layout from "../components/Layout";
+import { getAllData } from "../lib/icons";
+import useSearch from "../lib/search";
+
+function generateZip(icons) {
+  const zip = new JSZip();
+  Object.values(icons).forEach((icon) =>
+    // @ts-ignore
+    zip.file(`${icon.name}.svg`, icon.src)
+  );
+  return zip.generateAsync({ type: "blob" });
+}
+
+const IndexPage = ({ data }) => {
+  const [query, setQuery] = useQueryParam("query", StringParam);
+  const results = useSearch(data, query || "");
+  const toast = useToast();
+
+  const inputElement = useRef(null);
+  function handleKeyDown(event) {
+    if (event.key === "/" && inputElement.current !== document.activeElement) {
+      event.preventDefault();
+      inputElement.current.focus();
+    }
+  }
+  useEffect(() => {
+    window.addEventListener("keydown", handleKeyDown);
+    return () => window.removeEventListener("keydown", handleKeyDown);
+  }, []);
+
+  const [numColumns, setNumColumns] = useState(1);
+  useEffect(() => {
+    setNumColumns(Math.floor(window.outerWidth / 160));
+  }, []);
```

Comment:
I noticed that when page loads, all the icons where displayed in full-width for a second. I think we can fix this by calculating and setting it before render.
```suggestion
  const idealColumnCount = () => Math.floor(window.outerWidth / 160);
  const [numColumns, setNumColumns] = useState(idealColumnCount());
  useEffect(() => {
    setNumColumns(idealColumnCount());
  }, []);
```

Reply:
You can't use the `window` attribute outside of the `useEffect` 😕

Reply:
oh really, didn't know that.
But you can use It in useState right?

```suggestion
  const idealColumnCount = ({outerWidth}) => Math.floor(outerWidth / 160);
  const [numColumns, setNumColumns] = useState(idealColumnCount(window));
  useEffect(() => {
    setNumColumns(idealColumnCount(window));
  }, []);
```

Reply:
Nope for some reason that doesn't work either 🤔 I don't know why, this is very strange!

Reply:
If you want to get rid of the flash of one large column, I think you can avoid using JS to calculate the column width altogether.  Remove the `useEffect` and `useState` and instead change the grid css to be:

```diff
diff --git a/packages/site/src/pages/index.tsx b/packages/site/src/pages/index.tsx
index c31eb3d..6f3f04a 100644
--- a/packages/site/src/pages/index.tsx
+++ b/packages/site/src/pages/index.tsx
@@ -45,11 +45,6 @@ const IndexPage = ({ data }) => {
     return () => window.removeEventListener("keydown", handleKeyDown);
   }, []);
 
-  const [numColumns, setNumColumns] = useState(1);
-  useEffect(() => {
-    setNumColumns(Math.floor(window.outerWidth / 160));
-  }, []);
-
   return (
     <Layout>
       <Flex direction="column" align="center" justify="center">
@@ -81,7 +76,7 @@ const IndexPage = ({ data }) => {
       </InputGroup>
       {results.length > 0 ? (
         <Grid
-          templateColumns={`repeat(${numColumns > 5 ? 5 : numColumns}, 1fr)`}
+          templateColumns={`repeat(auto-fit, minmax(160px, 1fr))`}
           gap={5}
         >
           {results.map((icon) => {
```

 (sorry I don't know how to use the suggestion feature)

Reply:
@aelfric Thanks for your code snippet, it fixed the issue!